dev/release.py

```diff
@@ -0,0 +1,161 @@
+    
+    sha = open(file_name + ".sha").read().split(" ")[1]
+    return_str += "SHA: " + sha + "\n"
+
+    return_str += "Release artifacts are signed with the key. The KEYS file is available here:\n"
+    return_str += "\nhttps://dist.apache.org/repos/dist/release/incubator/reef/KEYS\n\n\n\n"
+
+    return_str += "<Issue Things>\n\n\n\n"
+
+    return_str += "The vote will be open for 72 hours. Please download the release\n" \
+        + "candidate, check the hashes/signature, build it and test it, and then\nplease vote:\n\n" \
+        + "[ ] +1 Release this package as Apache REEF " + reef_version + "\n" \
+        + "[ ] +0 no opinion\n" \
+        + "[ ] -1 Do not release this package because ...\n\n" \
+        + "Thanks!"
+
+    return return_str
+
+"""
+Function to exclude files in .gitignore when making tar.gz
+Return true when a file is in .gitignore
+"""
+def exclude_git_ignore(file_name):
+    ignore_list = get_ignore_list(reef_home)
+    for e in ignore_list:
+        if fnmatch.fnmatch(file_name, e):
+            return True
+
+    return fnmatch.fnmatch(file_name, "*/.git") or fnmatch.fnmatch(file_name, "*/.gitignore") or \
+        fnmatch.fnmatch(file_name, "*/.gitattributes")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Script for release versioning, packaging, checksum")
+    
+    parser.add_argument("reef_home", type=str, help="REEF home")
+    parser.add_argument("reef_version", type=str, help="REEF version")
+    parser.add_argument("rc_num", type=int, help ="rc candidate number")
+    parser.add_argument("key_id", type=str, help = "public key id (8 hex numbers)")
+    args = parser.parse_args()
+    
+    reef_home = os.path.abspath(args.reef_home)
+    reef_version = args.reef_version
+    rc_num = args.rc_num
+    key_id = args.key_id
+
+    change_version.change_build_props(reef_home + "/lang/cs/build.props", "false")
+    change_version.change_version(reef_home, reef_version)
+
+    build_result = subprocess.call("cd " + reef_home + " && " + "mvn apache-rat:check", shell=True)
+
+    if build_result == 0:
+        file_name = "apache-reef-" + reef_version + "-rc" + str(rc_num) + ".tar.gz"
+
+        # Make tar.gz
+        tar = tarfile.open(file_name, "w:gz")
+        tar.add(reef_home, arcname="apache-reef-"+reef_version , exclude=exclude_git_ignore)
+        tar.close()
+
+        gpg_str = "gpg --armor -u " + str(key_id) + " --output " + file_name + ".asc " + "--detach-sig " + file_name
+        gpg_result = subprocess.call(gpg_str, shell=True)
+
+        if gpg_result == 0:
+            md5 = hashlib.md5(open(file_name,'rb').read()).hexdigest()
+            sha = hashlib.sha512(open(file_name,'rb').read()).hexdigest()
+
+            md5_file = open(file_name + ".md5", "w")
+            md5_file.write("MD5(" + file_name + ")= " + md5 + "\n")
+            sha_file = open(file_name + ".sha", "w")
+            sha_file.write("SHA(" + file_name + ")= " + sha + "\n")
```

Comment:
If there are no objections, let's change to "coreutils style". This is the style [currently used on rc2](https://dist.apache.org/repos/dist/dev/incubator/reef/0.12.0-incubating-rc2/), the checksum followed by a space, asterisk, then the filename. (The asterisk means that the file being checksum-ed was treated as binary, which is what we want.)